static/shared/js/typeahead.js

```diff
@@ -93,39 +94,48 @@ export function get_emoji_matcher(query) {
 
 export function triage(query, objs, get_item = (x) => x) {
     /*
-        We split objs into four groups:
+        We split objs into three groups:
 
             - entire string exact match
-            - match prefix exactly with `query`
-            - match prefix case-insensitively
+            - word boundary matches
             - other
 
-        Then we concat the first three groups into
+        Then we concat the first two groups into
         `matches` and then call the rest `rest`.
     */
 
     const exactMatch = [];
-    const beginswithCaseSensitive = [];
-    const beginswithCaseInsensitive = [];
     const noMatch = [];
     const lowerQuery = query ? query.toLowerCase() : "";
+    const word_separator_regex = /[\s/_-]/;
+    let query_terms = lowerQuery.split(",");
+    query_terms = query_terms.map((s) => s.trim());
+
+    const wordBoundaryMatch = objs.filter((item) =>
+        query_terms.some((search_term) => {
+            const lower_name = get_item(item).toLowerCase();
+            // returns true if the item starts with the search term or if the
+            // search term with a word separator right before it appears in the item
+            return (
+                lower_name.startsWith(search_term) ||
+                new RegExp(word_separator_regex.source + _.escapeRegExp(search_term)).test(
```

Comment:
I don't think we should import lodash into this file, so will have to come up with an alternate way to do the escaping I think... Or maybe we could check for each of the separators (space, slash, hypen, underscore) if that concatenated with the query occurs in the string (thoughts, @timabbott?)

Reply:
Yes, it'd be better to avoid lodash, though @gnprice might be able to say to what extent it matters.

Reply:
Should I write a function that does the escaping?